src/NoFrixion.MoneyMoov/Claims/IdentityExtensions.cs

```diff
@@ -299,6 +300,31 @@ public static string WhoAmI(this ClaimsIdentity claimsIdentity)
             }
         }
     }
+
+    /// <summary>
+    /// Returns true if the identity has the specified permission for the merchant.
+    /// Else returns false.
+    /// </summary>
+    /// <param name="identity">The token identity</param>
+    /// <param name="permissions">The permissions to be searched in claims.</param>
+    /// <param name="merchantID">The ID of the merchant to look permissions for.</param>
+    /// <returns></returns>
+    public static bool HasMerchantPermissions(this IIdentity identity, MerchantPermissions permissions, Guid merchantID)

```

Comment:
Can you call this HasMerchantPermission (singular) instead?

Reply:
@donalnofrixion I wanted it to be singular but since it's a flags enum, I think one can pass multiple permissions. So I made it plural. As in this method can work if you wanted to check multiple merchant permissions.

Reply:
Let me try that.

Reply:
You're right. I'll have to change it to list of permissions for multiple. I'll make it singular.